### PR TITLE
WIP: Add permission check for some services

### PR DESCRIPTION
**Pull request by vinothvino42**
Ticket: #701

1. Notification (Added by vu)
2. Location
3. Contacts
4. Camera

### CODE REVIEW

File: lib/framework/permissions_manager.dart [hunk 1]
```diff
@@ -11,33 +14,63 @@ class PermissionsManager {
...
+            case AuthorizationStatus.provisional:
+              status = true;
+              break;
+            case AuthorizationStatus.denied:
+              status = false;
+              break;
+            case AuthorizationStatus.notDetermined:
+            default:
+              status = null;
+              break;
```
Comment (evshi) [inline lib/framework/permissions_manager.dart:38]: We should move this location specific logic into the utilities or this class will become very large.

**Review by evshi — APPROVED** [review]

File: lib/framework/stub/camera_manager.dart [hunk 2]
```diff
@@ -4,11 +4,18 @@ import 'package:ensemble/framework/scope.dart';
...
 abstract class CameraManager {
+  Future<bool?> hasPermission();
   Future<void> openCamera(BuildContext context, ShowCameraAction cameraAction,
       ScopeManager? scopeManager);
 }
 
 class CameraManagerStub extends CameraManager {
+  @override
+  Future<bool?> hasPermission() {
+    throw ConfigError(
```
Comment (vusters) [inline lib/framework/stub/camera_manager.dart:15]: We don't throw an Error here, simply false. This is valid use case. We only throw the error if the user attempts to use the camera.

Comment (vinothvino42) [reply lib/framework/stub/camera_manager.dart]: I updated it. Can you check now?

File: lib/framework/stub/contacts_manager.dart [hunk 3]
```diff
@@ -20,8 +20,7 @@ class ContactManagerStub extends ContactManager {
 
   @override
   Future<bool> requestPermission() {
-    throw ConfigError(
-        "Phone Contact Service is not enabled. Please review the Ensemble documentation.");
+    return Future.value(false);
```
Comment (vusters) [inline lib/framework/stub/contacts_manager.dart:23]: is this checking for permission or request user access? If it's the latter you'd want to throw the Error right?

Comment (vinothvino42) [reply lib/framework/stub/contacts_manager.dart:23]: According to the doc, there's no dedicated method to check for the permission is enabled or not in the flutter_contacts package.

FYI, See the following
1. https://pub.dev/documentation/flutter_contacts/latest/flutter_contacts/FlutterContacts-class.html
2.https://github.com/QuisApp/flutter_contacts/blob/b1fd688c34968c30e76977479e4ed392bcb3bfda/ios/Classes/SwiftFlutterContactsPlugin.swift#L447C15-L447C32

There's a way to do that natively, But this package didn't expose the method `authorizationStatus(for:)` to check the permission which returns `CNAuthorizationStatus` - https://developer.apple.com/documentation/contacts/cncontactstore/1403143-authorizationstatus

**Review by vusters — APPROVED** [review]

**Pull request merged**